src/content/app/genome-browser/hooks/useGenomicTracks.ts

```diff
@@ -0,0 +1,173 @@
+
+import { getAllTrackSettings } from 'src/content/app/genome-browser/state/track-settings/trackSettingsSelectors';
+import { getBrowserTrackStates } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
+
+import { Status } from 'src/shared/types/status';
+import type { GenomeTrackCategory } from 'src/content/app/genome-browser/state/types/tracks';
+import type {
+  TrackStates,
+  TrackActivityStatus
+} from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
+import type { TrackSettings } from 'src/content/app/genome-browser/state/track-settings/trackSettingsSlice';
+
+/**
+ * The purposes of this hook are:
+ * - tell the genome browser what tracks (apart from the focus track, which is
+ *   another hook’s responsibility) to show for any given genome id
+ * - tell the genome browser which settings to apply to the shown tracks
+ *
+ * Only one copy of this hook should be run.
+ */
+
+const useGenomicTracks = () => {
+  const { activeGenomeId } = useGenomeBrowserIds();
+  const allBrowserTrackStates = useAppSelector(getBrowserTrackStates);
+  const trackSettingsForGenome =
+    useAppSelector(getAllTrackSettings)?.tracks ?? {};
+  const { genomeBrowser, ...genomeBrowserMethods } = useGenomeBrowser();
+
+  const { data: genomeTrackCategories } = useGenomeTracksQuery(
+    activeGenomeId as string,
+    {
+      skip: !activeGenomeId
+    }
+  );
+
+  const savedGenomicTrackStates =
+    allBrowserTrackStates[activeGenomeId ?? '']?.commonTracks ?? {};
+
+  useEffect(() => {
+    if (!genomeBrowser) {
+      return;
+    }
+
+    const trackIdsList = combineWithSavedData(
+      prepareTrackIdsList(genomeTrackCategories ?? []),
+      savedGenomicTrackStates
+    );
+    trackIdsList.forEach(genomeBrowserMethods.toggleTrack);
+  }, [
+    activeGenomeId,
+    genomeTrackCategories,
+    savedGenomicTrackStates,
+    genomeBrowser
+  ]);
+
+  useEffect(() => {
+    if (!genomeBrowser) {
+      return;
+    }
+
+    sendTrackSettings(trackSettingsForGenome, genomeBrowserMethods);
+  }, [trackSettingsForGenome, genomeBrowser]);
+};
+
+type TrackIdsList = {
+  trackId: string;
+  status: Status;
+}[];
+
+// Create a list of tracks to enable in the genome browser.
+// Assume that all tracks should be enabled by default
+const prepareTrackIdsList = (
+  trackGroups: GenomeTrackCategory[]
+): TrackIdsList => {
+  return trackGroups
+    .flatMap(({ track_list }) => track_list)
+    .map(({ track_id }) => ({
+      trackId: track_id,
+      status: Status.SELECTED
+    }));
+};
+
+const combineWithSavedData = (
+  trackIdsList: TrackIdsList,
+  savedTrackStates: TrackStates
+): TrackIdsList => {
+  const savedTrackVisibilityMap = new Map<string, TrackActivityStatus>();
+  Object.values(savedTrackStates)
+    .flatMap((trackCategory) => Object.entries(trackCategory))
+    .forEach(([trackId, trackStatus]) => {
+      savedTrackVisibilityMap.set(trackId, trackStatus);
+    }, {});
```

Comment:
Why do we need `, {}`?

Reply:
It's leftover of me writing a reduce first 😄 Surprised that it's valid for the `forEach`; at least typescript swallowed it.

Reply:
✅